### PR TITLE
Modal

### PR DESCRIPTION
**Pull request by alschmiedt**
Separate Modal from Typed Variable Modal

### CODE REVIEW

File: plugins/modal/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  toolbox: toolboxCategories,
+});
+
+// Initialize plugin.
+const modal = new Modal(workspace);
+modal.init();
+```
+
+## API
+- `init`: Create a  modal.
```
Comment (samelhusseini) [inline plugins/modal/README.md]: nit: remove extra space.

File: plugins/modal/README.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+- `dispose`: Dispose of the modal.
+- `show`: Shows the modal and focus on the first focusable element.
```
Comment (samelhusseini) [inline plugins/modal/README.md]: Show the modal and focus on the first interactable element.

File: plugins/modal/README.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+- `hide`: Hide the modal.
+- `render`: Create all the dom elements for the modal.
+- `shouldCloseOnOverlayClick`: (default true) If set to true will close the
```
Comment (samelhusseini) [inline plugins/modal/README.md:42]: It's not clear what's a call and whats an option from this API.

File: plugins/modal/package.json [hunk 4]
```diff
@@ -0,0 +1,68 @@
+{
+  "name": "@blockly/plugin-modal",
+  "version": "0.20200427.0",
```
Comment (samelhusseini) [inline plugins/modal/package.json:3]: Use 1.x unless it's in beta.

File: plugins/modal/package.json [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  "description": "A Blockly plugin that creates a modal.",
```
Comment (samelhusseini) [inline plugins/modal/package.json:4]: A Blockly plugin that renders a modal.

File: plugins/modal/package.json [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    "lint": "blockly-scripts lint",
+    "prepublishOnly": "npm run clean && npm run build",
+    "start": "blockly-scripts start",
+    "test": "npm run build && npm run test:mocha",
+    "test:mocha": "blockly-scripts test"
+  },
+  "main": "./dist/index.js",
+  "module": "./src/index.js",
+  "unpkg": "./dist/index.js",
+  "author": "",
```
Comment (samelhusseini) [inline plugins/modal/package.json]: Add Author

File: plugins/modal/test/index.html [hunk 7]
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
+<html>
+
+<head>
+  <meta charset="utf-8">
+  <title>Blockly Plugin Test</title>
```
Comment (samelhusseini) [inline plugins/modal/test/index.html]: Blockly Modal Test

File: plugins/modal/test/index.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+import {Modal} from '../src/index';
+
+/**
+ * Test page startup, sets up Blockly.
+ */
+function start() {
+  const workspace = Blockly.inject('blocklyDiv', {
+    toolbox: toolboxCategories,
+  });
+  console.log(Modal);
```
Comment (samelhusseini) [inline plugins/modal/test/index.js]: Remove console

File: plugins/modal/test/index.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Plugin test.
```
Comment (samelhusseini) [inline plugins/modal/test/index.js]: Modal test

**Review by samelhusseini — COMMENTED** [review]
Added comments, but where is src/index.js?

**Comment by alschmiedt** [comment]
plugins/typed-variable-modal/src/Modal.js → plugins/modal/src/index.js
It looks like a rename instead of an add since I just copied the modal file over from typedvariablemodal

**Comment by samelhusseini** [comment]
Oh yep I see it.

**Review by samelhusseini — APPROVED** [review]
LGTM once comments are addressed.

File: plugins/modal/README.md (hunk 2 above)
Comment (alschmiedt) [reply plugins/modal/README.md]: done

File: plugins/modal/package.json (hunk 4 above)
Comment (alschmiedt) [reply plugins/modal/package.json:3]: What about leaving this as zero and bump when we publish?

File: plugins/modal/package.json (hunk 5 above)
Comment (alschmiedt) [reply plugins/modal/package.json:4]: done

File: plugins/modal/package.json (hunk 6 above)
Comment (alschmiedt) [reply plugins/modal/package.json]: done

File: plugins/modal/test/index.html (hunk 7 above)
Comment (alschmiedt) [reply plugins/modal/test/index.html]: done

File: plugins/modal/test/index.js (hunk 9 above)
Comment (alschmiedt) [reply plugins/modal/test/index.js]: done

File: plugins/modal/test/index.js (hunk 8 above)
Comment (alschmiedt) [reply plugins/modal/test/index.js]: done

File: plugins/modal/README.md (hunk 1 above)
Comment (alschmiedt) [reply plugins/modal/README.md]: done

File: plugins/modal/package.json (hunk 4 above)
Comment (samelhusseini) [reply plugins/modal/package.json:3]: Sure.

**Review by samelhusseini — APPROVED** [review]

**Pull request merged**